librz/include/rz_core.h

```diff
@@ -402,6 +402,7 @@ RZ_API void rz_core_notify_error(RZ_NONNULL RzCore *core, RZ_NONNULL const char
  */
 RZ_API void rz_core_gadget_free(RzCoreGadget *g);
 RZ_API void rz_core_gadget_print(RzCore *core);
+RZ_API void rz_core_debug_switch_to_first_thread(RzCore *core);
```

Comment:
i wonder if this should be part of `rz_debug`.

Reply:
Yeah makes since I'm mostly using the debug API's inside this function, will fix.

Reply:
I'm trying to test it on this piece of multithreading code. But When I try to list the active threads that are attached to this process, it only lists one. 

```C
#include <stdio.h>
#include <pthread.h>
#include <unistd.h>
#include <sys/types.h>

#define NUM_THREADS 5

void *thread_function(void *arg) {
    printf("Thread %ld is running...\n", (long)arg);
    printf("Thread ID is %lu\n", (unsigned long)pthread_self());
    sleep(10);
    printf("Thread %ld finished.\n", (long)arg);
    pthread_exit(NULL);
}

int main() {
    pthread_t threads[NUM_THREADS];
    int rc;
    long t;

    for(t = 0; t < NUM_THREADS; t++) {
        printf("Creating thread %ld\n", t);
        rc = pthread_create(&threads[t], NULL, thread_function, (void *)t);
        if (rc) {
            printf("ERROR; return code from pthread_create() is %d\n", rc);
            return -1;
        }
    }

    for(t = 0; t < NUM_THREADS; t++) {
        pthread_join(threads[t], NULL);
    }

    return 0;
}
```





```sh
$ sudo rizin -d test
Process with PID 147589 started...
WARNING: Neither hash nor gnu_hash exist. Falling back to heuristics for deducing the number of dynamic symbols...
WARNING: Neither hash nor gnu_hash exist. Falling back to heuristics for deducing the number of dynamic symbols...
WARNING: Neither hash nor gnu_hash exist. Falling back to heuristics for deducing the number of dynamic symbols...
 -- Review all the subcommands of aa to see better ways to analyze your targets.
[0x7f330c02bbc0]> dp
 * 147589 ppid:147587 uid:0 s ./test
[0x7f330c02bbc0]> dpt
current ppid   pid    uid status path   
----------------------------------------
true    147587 147589 0   s      ./test
[0x7f330c02bbc0]> 
```